include/circular_queue.h

```diff
@@ -0,0 +1,58 @@
+#ifndef QUEUE_GUARD
+#define QUEUE_GUARD
+
+#include <condition_variable>
+#include <mutex>
+#include <utility>
+
+struct queue_elm {
+	bool dirty;    // is this queue element yet to be processed by sketching (if so do not overwrite)
+	uint32_t size; // the size of this data element
+	char *data;    // the data element which goes in the queue
```

Comment:
does C++ auto-reorder struct fields? if it's like C, we'll want to re-order these to save space

Reply:
It probably does reorder but there's no harm in doing it ourselves. I don't quite know what you mean though because char * is either int32 or int64 if the system is 32 or 64 bit respectively. So I can't see how we would save space via reordering. Am I missing something?

Reply:
Ah you're right. I thought it was a char